apisix/plugins/hmac-auth.lua

```diff
@@ -193,6 +206,19 @@ local function do_nothing(v)
     return v
 end
 
+local function validate_body(ctx, secret_key, params, req_body)
+    if not req_body then
+        req_body = ""
+    end
+    local digest_header = core.request.header(ctx, DIGEST)
```

Comment:
`digest` header looks not good for me.

Reply:
The reason I use this header is that:
The Digest response HTTP header provides a digest of the requested resource.